servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultMultiAddressUrlHttpClientBuilder.java

```diff
@@ -263,21 +265,32 @@ public StreamingHttpClient apply(final UrlKey urlKey) {
             this.executionContext = requireNonNull(executionContext);
         }
 
-        private FilterableStreamingHttpClient selectClient(
-                HttpRequestMetaData metaData) {
-            return group.get(keyFactory.apply(metaData));
+        private FilterableStreamingHttpClient selectClient(HttpRequestMetaData metaData) throws MalformedURLException {
+            return group.get(keyFactory.get(metaData));
         }
 
         @Override
         public Single<? extends FilterableReservedStreamingHttpConnection> reserveConnection(
                 final HttpExecutionStrategy strategy, final HttpRequestMetaData metaData) {
-            return defer(() -> selectClient(metaData).reserveConnection(strategy, metaData).subscribeShareContext());
+            return defer(() -> {
+                try {
+                    return selectClient(metaData).reserveConnection(strategy, metaData).subscribeShareContext();
+                } catch (Throwable t) {
+                    return Single.<FilterableReservedStreamingHttpConnection>failed(t).subscribeShareContext();
+                }
+            });
         }
 
         @Override
         public Single<StreamingHttpResponse> request(final HttpExecutionStrategy strategy,
                                                      final StreamingHttpRequest request) {
-            return defer(() -> selectClient(request).request(strategy, request).subscribeShareContext());
+            return defer(() -> {
+                try {
+                    return selectClient(request).request(strategy, request).subscribeShareContext();
+                } catch (Throwable t) {
```

Comment:
The `try-catch` here is necessary anyway, because methods that return async sources are not expected to throw.